File: iosApp/iosApp/Presentation/Login/Mapper/SmartSaverTransactionDomainToPresentationModelMapper.swift
//
//  SmartSaverTransactionDomainToPresentationModelMapper.swift
//  iosApp
//
//  Created by Animasahun Ibrahim on 14/03/2024.
//  Copyright © 2024 orgName. All rights reserved.
//

import Foundation
import IosData

class SmartSaverTransactionDomainToPresentationModelMapper: DomainToPresentationMapper{
    typealias T = SmartSaverTransactionDomainModel
    
    typealias U = SmartSaverTransactionPresentationModel
    
    func toPresentation(input: SmartSaverTransactionDomainModel) -> SmartSaverTransactionPresentationModel {
        
        return .init(
            amount: input.amount,
            type: input.type,
            narration: input.narration,
            dateCreated: input.dateCreated
        )
    }
    
}

File: app-presentation/src/main/java/com/teewhydope/app/presentation/login/mapper/SmartSaverTransactionDomainToPresentationModelMapper.kt
package com.teewhydope.app.presentation.login.mapper

import com.teewhydope.app.domain.authentication.model.SmartSaverTransactionDomainModel
import com.teewhydope.app.presentation.common.mapper.DomainToPresentationMapper
import com.teewhydope.app.presentation.login.model.SmartSaverTransactionPresentationModel

class SmartSaverTransactionDomainToPresentationModelMapper :
    DomainToPresentationMapper<SmartSaverTransactionDomainModel, SmartSaverTransactionPresentationModel>() {
    override fun map(input: SmartSaverTransactionDomainModel) = SmartSaverTransactionPresentationModel(
        amount = input.amount,
        type = input.type,
        narration = input.narration,
        dateCreated = input.dateCreated,
    )
}